src/containers/flows/AcceptBid/AcceptBid.tsx

```diff
@@ -0,0 +1,278 @@
+					account,
+				);
+				// Timeout to prevent jolt
+				await new Promise((r) => setTimeout(r, 1500));
+				if (isApproved) {
+					setStepTitle(TITLES[Step.Confirmation].PRIMARY);
+					setCurrentStep(Step.Confirmation);
+				} else {
+					setCurrentStep(Step.ApproveZAuction);
+				}
+			} catch (e) {
+				setError(ERRORS.CONSOLE_TEXT);
+				setCurrentStep(Step.Confirmation);
+				console.error(ERRORS.CONSOLE_TEXT, e);
+			}
+		})();
+	};
+
+	// Approve zAuction Flow
+	const approveZAuction = () => {
+		if (!sdk || !library || !account) {
+			return;
+		}
+		setError(undefined);
+		setCurrentStep(Step.WaitingForWallet);
+		(async () => {
+			try {
+				const zAuction = await sdk.getZAuctionInstanceForDomain(domainId);
+				const tx = await zAuction.approveZAuctionTransferNft(
+					library.getSigner(),
+				);
+				try {
+					setCurrentStep(Step.ApprovingZAuction);
+					await tx.wait();
+				} catch (e) {
+					setCurrentStep(Step.ApproveZAuction);
+					setError(ERRORS.TRANSACTION);
+				}
+				setStepTitle(TITLES[Step.Confirmation].PRIMARY);
+				setCurrentStep(Step.Confirmation);
+			} catch (e) {
+				setCurrentStep(Step.ApproveZAuction);
+				setError(ERRORS.REJECTED_WALLET);
+			}
+		})();
+	};
+
+	const onDetailsAccept = () => {
+		setCurrentStep(Step.CheckingZAuctionApproval);
+		setStepTitle(TITLES[Step.CheckingZAuctionApproval].PRIMARY);
+		checkZAuctionApproval();
+	};
+
+	const onBack = () => {
+		setError(undefined);
+		setCurrentStep(Step.Details);
+	};
+
+	const onConfirm = async () => {
+		setCurrentStep(Step.Accepting);
+		setStepTitle(TITLES[Step.Details].PRIMARY);
+		try {
+			await accept(acceptingBid!);
+			setStepTitle(TITLES[Step.Success].PRIMARY);
+			setCurrentStep(Step.Success);
+		} catch (e) {
+			setError(e.message);
+			setCurrentStep(Step.Confirmation);
+		}
+		if (!isMounted.current) return;
+	};
+
+	const onFinish = async () => {
+		if (acceptingBid && refetch) {
+			refetch(acceptingBid);
+			onClose();
+		}
+	};
+
+	const confirmationErrorButtonText = () =>
+		error
+			? BUTTONS[Step.Confirmation].TERTIARY
+			: BUTTONS[Step.Confirmation].PRIMARY;
+
+	/////////////
+	// Effects //
+	/////////////
+
+	// Set initial step
+	useEffect(() => {
+		if (isLoading) {
+			setCurrentStep(Step.LoadingData);
+		} else {
+			setCurrentStep(Step.Details);
+		}
+	}, [isLoading]);
+
+	useEffect(() => {
+		isMounted.current = true;
+		return () => {
+			isMounted.current = false;
+		};
+	});
+
+	const steps = {
+		// Loading Data
+		[Step.LoadingData]: <Wizard.Loading message={MESSAGES.TEXT_LOADING} />,
+		// Display NFT details
+		[Step.Details]: acceptingBid ? (
+			<Details
+				assetUrl={assetUrl}
+				creator={creatorId}
+				domainName={domainName}
+				title={domainTitle}
+				walletAddress={walletAddress}
+				bidAmount={acceptingBid.amount}
+				highestBid={highestBid}
+				wildPriceUsd={wildPriceUsd}
+				onClose={onClose}
+				onNext={() => onDetailsAccept()}
```

Comment:
nit: should be sweet to just do `onNext={onDetailsAccept}` (same as in a couple of other spots) rather than using an anonymous function